python_modules/dagster/dagster/_core/definitions/decorators/assets_definition_factory.py

```diff
@@ -0,0 +1,52 @@
+from functools import cached_property
+from typing import Mapping, NamedTuple, Tuple
+
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.input import In
+from dagster._core.definitions.output import Out
+
+
+class InMapping(NamedTuple):
```

Comment:
Realized that this name is an almost-collision with `InputMapping`, which is something very different. Thoughts on naming this something like `NamedIn`?

Reply:
Addressed in https://github.com/dagster-io/dagster/pull/22251